webclient/src/pages/users/Signup.js

```diff
@@ -1,87 +1,57 @@
-  };
+  });
 
-  handleReload = e => {
-    window.location.replace('/');
-  };
+  const authClient = useAuth();
 
-  handleChange = e => {
-    this.setState({
+  const handleChange = e => {
+    setState({
+      ...state,
       [e.target.id]: e.target.value
     });
   };
 
-  handleSubmit = e => {
+  const handleSubmit = e => {
     e.preventDefault();
-    const awaitSignUp = new Promise((resolve, reject) => {
-      this.props.signUpUser(this.state);
-      setTimeout(() => resolve('Sign Up Success'), 500);
-    });
-    awaitSignUp.then(value => {
-      this.handleReload();
-    });
+    authClient.register(state);
   };
 
-  render() {
-    const { auth, authError } = this.props;
-    //if (auth.uid) return <Redirect to='/' />
-
-    return (
-      <div className="container">
-        <form className="grey lighten-4" onSubmit={this.handleSubmit}>
-          <h5 className="grey-text text-darken-3">Sign Up</h5>
-          <div className="input-field">
-            <label htmlFor="email">Email</label>
-            <input type="email" id="email" onChange={this.handleChange} />
-          </div>
-
-          <div className="input-field">
-            <label htmlFor="firstName">FirstName</label>
-            <input type="text" id="firstName" onChange={this.handleChange} />
-          </div>
-
-          <div className="input-field">
-            <label htmlFor="lastName">LastName</label>
-            <input type="text" id="lastName" onChange={this.handleChange} />
-          </div>
-
-          <div className="input-field">
-            <label htmlFor="password">Password</label>
-            <input type="password" id="password" onChange={this.handleChange} />
-          </div>
-          <div className="input-field">
-            <button className="btn z-depth-0">Sign Up</button>
-            <div className="red-text center">
-              {authError ? <p>{authError}</p> : null}
-            </div>
-          </div>
-        </form>
-      </div>
-    );
-  }
+  return (
+    <div className="container">
+      <form className="grey lighten-4" onSubmit={handleSubmit}>
+        <h5 className="grey-text text-darken-3">Sign Up</h5>
+        <div className="input-field">
+          <label htmlFor="email">Email</label>
+          <input type="email" id="email" onChange={handleChange} />
+        </div>
+
+        <div className="input-field">
+          <label htmlFor="firstName">FirstName</label>
+          <input type="text" id="firstName" onChange={handleChange} />
+        </div>
+
+        <div className="input-field">
+          <label htmlFor="lastName">LastName</label>
+          <input type="text" id="lastName" onChange={handleChange} />
+        </div>
+
+        <div className="input-field">
+          <label htmlFor="password">Password</label>
+          <input type="password" id="password" onChange={handleChange} />
+        </div>
+        <div className="input-field">
+          <button className="btn z-depth-0">Sign Up</button>

```

Comment:
shouldn't this be `<input type=submit ...>`

Reply:
ok changed